common/component/kafka/metadata.go

```diff
@@ -307,5 +323,26 @@ func (k *Kafka) getKafkaMetadata(meta map[string]string) (*KafkaMetadata, error)
 		m.consumerFetchMin = int32(v)
 	}
 
+	// producer connection specifications
+	if val, ok := meta[clientConnectionRefreshIntervalMin]; ok && val != "" {
+		v, err := strconv.Atoi(val)
+		if err != nil {
+			return nil, err
+		}
+		m.clientConnectionRefreshIntervalMin = v
+	} else {
+		m.clientConnectionRefreshIntervalMin = defaultClientConnectionRefreshIntervalMin
+	}
+
+	if val, ok := meta[clientConnectionMaxKeepAliveMin]; ok && val != "" {
+		v, err := strconv.Atoi(val)
+		if err != nil {
+			return nil, err
+		}
+		m.clientConnectionMaxKeepAliveMin = v
+	} else {
+		m.clientConnectionMaxKeepAliveMin = defaultClientConnectionMaxKeepAliveMin
+	}
+
```

Comment:
This entire block can be removed if you use `time.Duration` as I suggested for the properties...


Although in that case you may want to add some validation:

- Ensure that clientConnectionRefreshInterval is > 0
- Ensure that clientConnectionKeepAliveInterval is >= 0 (and if it's < 0, make it = 0)